### PR TITLE
feat(financekit): check authorization status at start of all functions

### PR DESCRIPTION
**Pull request by nikwotton**

### CODE REVIEW

File: packages/npm/financekit/src/index.tsx [hunk 1]
```diff
@@ -342,6 +344,7 @@ export function transactions(
                   '@candlefinance.financekit.query_invalid',
                   '@candlefinance.financekit.unknown',
                   '@candlefinance.financekit.transactions_invalid',
+                  '@candlefinance.financekit.unknown_error_response_schema',
```
Comment (liambutler-lawrence) [inline packages/npm/financekit/src/index.tsx:347]: Is this supposed to be `'@candlefinance.financekit.unauthorized_financekit_usage'`? Same for rest of functions

Comment (nikwotton) [reply packages/npm/financekit/src/index.tsx:347]: 🤦 copied the wrong one...

File: packages/npm/financekit/ios/Financekit.swift [hunk 2]
```diff
@@ -8,12 +8,14 @@ private let CODE_PARAMS_INVALID = "@candlefinance.financekit.params_invalid"
...
 private let CODE_TRANSACTION_HISTORY_INVALID = "@candlefinance.financekit.transaction_history_invalid"
 private let CODE_TRANSACTIONS_INVALID = "@candlefinance.financekit.transactions_invalid"
+private let CODE_UNAUTHORIZED_USAGE = "@candlefinance.financekit.unauthorized_financekit_usage"
 private let CODE_UNKNOWN = "@candlefinance.financekit.unknown"
 private let CODE_UNKNOWN_ACCOUNT_TYPE = "@candlefinance.financekit.unknown_account_type"
 private let CODE_UNKNOWN_AUTHORIZATION_STATUS = "@candlefinance.financekit.unknown_authorization_status"
 private let CODE_UNKNOWN_CURRENT_BALANCE_TYPE = "@candlefinance.financekit.unknown_current_balance_type"
 
 private let MESSAGE_LOW_OS_VERSION = "You are trying to use FinanceKit but your phone's OS is not 17.4.0 or above"
+private let MESSAGE_UNAUTHORIZED_USAGE = "You are trying to use FinanceKit but you did not have the user authorize you to do so first"
```
Comment (liambutler-lawrence) [inline packages/npm/financekit/ios/Financekit.swift:18]: This message is only accurate if the status is notDetermined fwiw

**Review by liambutler-lawrence — DISMISSED** [review]

File: packages/npm/financekit/src/index.tsx (hunk 1 above)
Comment (nikwotton) [reply packages/npm/financekit/src/index.tsx]: fixed

File: packages/npm/financekit/ios/Financekit.swift (hunk 2 above)
Comment (nikwotton) [reply packages/npm/financekit/ios/Financekit.swift]: Tweaked the message...better now?

**Review by liambutler-lawrence — APPROVED** [review]

**Pull request merged**